code/modules/paperwork/paper_bundle.dm

```diff
@@ -1,79 +1,178 @@
+#define MAX_PHOTO_OVERLAYS  10  //Maximum amount of photo overlays displayed on the paper bundle
+#define MAX_PAPER_UNDERLAYS 20  //Maximum amount of paper underlays displayed under the bundle icon
+
+///////////////////////////////////////////////////////////////////////////
+// Paper Bundle
+///////////////////////////////////////////////////////////////////////////
 /obj/item/paper_bundle
-	name = "paper bundle"
-	gender = NEUTER
-	icon = 'icons/obj/bureaucracy.dmi'
-	icon_state = "paper"
-	item_state = "paper"
-	randpixel = 8
-	throwforce = 0
-	w_class = ITEM_SIZE_SMALL
-	throw_range = 2
-	throw_speed = 1
-	layer = ABOVE_OBJ_LAYER
-	attack_verb = list("bapped")
-	health = 10
-	max_health = 10
-	var/page = 1    // current page
-	var/list/pages = list()  // Ordered list of pages as they are to be displayed. Can be different order than src.contents.
+	name              = "paper bundle"
+	icon              = 'icons/obj/bureaucracy.dmi'
+	icon_state        = "paper"
+	item_state        = "paper"
+	layer             = ABOVE_OBJ_LAYER
+	randpixel         = 8
+	throwforce        = 0
+	throw_range       = 2
+	throw_speed       = 1
+	w_class           = ITEM_SIZE_SMALL
+	attack_verb       = list("bapped")
+	drop_sound        = 'sound/foley/paperpickup1.ogg'
+	pickup_sound      = 'sound/foley/paperpickup2.ogg'
+	item_flags        = ITEM_FLAG_CAN_TAPE
+	health            = 10
+	max_health        = 10
+	var/tmp/cur_page  = 1           // current page
+	var/tmp/max_pages = 100         //Maximum number of papers that can be in the bundle
+	var/list/pages                  // Ordered list of pages as they are to be displayed. Can be different order than src.contents.
+	var/static/list/cached_overlays //Cached images used by all paper bundles for generating the overlays and underlays
+
+/**Creates frequently used images globally, so we can re-use them. */
```

Comment:
Again, if you do this right SSoverlays will cache them for you. Even if not, it's not worth caching images.

Reply:
Paper bundles are a bit different though. Pretty sure there's a gain to using only a single source icon when used with mutable appearance.

Reply:
Run it by Loaf or Lohikar; image manipulation is not very expensive.